### PR TITLE
Replace deprecated JObject with new stdClass()

### PR DESCRIPTION
**Pull request by C-Lodder**
### Summary of Changes

`JObject` is deprecated and will be removed in J4.x. This PR removes the majority of references to `JObject` in favour of `new stdClass()`.

**Note:**
I haven't touched class extending, such as `JMenuNode extends JObject` as again, not sure how this would be done.
### Testing Instructions

Code review.
### Documentation Changes Required

None

### CODE REVIEW

**Comment by mbabker** [comment]
> I also haven't touched class extending, such as JMenuNode extends JObject as again, not sure how this would be done.

Has to wait for 4.0 as it'd be a B/C break to stop being able to use the API methods it provides.

**Comment by mbabker** [comment]
The other thing to think of here is `stdClass` doesn't allow you to have object keys with dots in them.  Like where we create a `JObject` to check permissions in components where you have something like `$object->set('core.create', $value);`.  In those cases you'd be better off using a `Registry` object instead.

**Comment by C-Lodder** [comment]
@mbabker - ah ok that's fine. I've now removed the reference in `$foo = JArrayHelper::toObject($bar, 'JObject');` too

File: administrator/components/com_media/controllers/file.json.php [hunk 1]
```diff
@@ -111,7 +111,7 @@ public function upload()
 		// Trigger the onContentBeforeSave event.
 		JPluginHelper::importPlugin('content');
 		$dispatcher  = JEventDispatcher::getInstance();
-		$object_file = new JObject($file);
+		$object_file = new stdClass($file);
```
Comment (mbabker) [inline administrator/components/com_media/controllers/file.json.php]: This probably has to wait for 4.0 as it changes the object that's passed via the plugin event a couple of lines down.

File: administrator/components/com_media/controllers/folder.php [hunk 2]
```diff
@@ -83,7 +83,7 @@ public function delete()
 				}
 
 				$fullPath = JPath::clean(implode(DIRECTORY_SEPARATOR, array(COM_MEDIA_BASE, $folder, $path)));
-				$object_file = new JObject(array('filepath' => $fullPath));
+				$object_file = new stdClass(array('filepath' => $fullPath));
```
Comment (mbabker) [inline administrator/components/com_media/controllers/folder.php]: `stdClass` doesn't have a constructor.

File: administrator/components/com_plugins/helpers/plugins.php [hunk 3]
```diff
@@ -97,11 +97,11 @@ public static function folderOptions()
 	 * @param   string  $templateBaseDir  Base path to the template directory.
 	 * @param   string  $templateDir      Template directory.
 	 *
-	 * @return  JObject
+	 * @return  object
```
Comment (mbabker) [inline administrator/components/com_plugins/helpers/plugins.php]: This has to wait for 4.0; it changes the return type on a public API to a non-compatible object.

File: libraries/legacy/model/legacy.php [hunk 4]
```diff
@@ -53,7 +53,7 @@
 	/**
 	 * A state object
 	 *
-	 * @var    JObject
+	 * @var    object
```
Comment (mbabker) [inline libraries/legacy/model/legacy.php]: This will need to be a `Registry` object.

**Comment by C-Lodder** [comment]
@mbabker Just reverting some of the changes based on your comments. As for the last one...`This will need to be a Registry object.`, does that also mean changing this:
```
$this->state = new stdClass();
```
to this:
```
$this->state = new JRegistry();
```
?

**Comment by mbabker** [comment]
Yes.

**Comment by zero-24** [comment]
@mbabker JRegistry and not Registry?

**Comment by C-Lodder** [comment]
Hmm, I think this is causing too many problems for Travis. Close?

**Comment by mbabker** [comment]
> Hmm, I think this is causing too many problems for Travis. Close?

Do it in smaller chunks.  It's honestly a major project you're not going to be able to do in one big push just because there are so many uses of it.

**Comment by C-Lodder** [comment]
Ok I'll revisit this later on tonight or tomorrow morning

**Comment by wilsonge** [comment]
Given that closing this

**Pull request closed without merging**